test/spmd/test_tracing.py

```diff
@@ -410,5 +414,65 @@ def f(x, y):
         self._test_expand(xd, yd, f, mesh, out_placement)
 
 
+class TraceModuleTest(DistTensorTestBase):
+    @property
+    def world_size(self):
+        return 2
+
+    def _test_trace_replicate(self, model, x):
+        # if x.device.type == "cuda":
+        ddp = DDP(deepcopy(model))
+        spmd = SPMD(
+            deepcopy(model),
+            schema=Schema(
+                mesh=DeviceMesh(
+                    self.device_type, torch.arange(self.world_size)
+                ),
+                placements=[Replicate()],
+            ),
+        )
+
+        ddp(x).sum().backward()
+        spmd(x).sum().backward()
+        for p1, p2 in zip(ddp.parameters(), spmd.parameters()):
+            # DDP divides gradients by world size to compute average, but
+            # _Partial tensor shouldn't do that automatically. Hence explicitly
+            # do division here.
+            self.assertTrue(p1.grad.allclose(p2.grad / self.world_size))
+
+    @with_comms
+    def test_sequential(self):
+        if self.device_type == "cuda":
+            device = torch.device(self.rank)
+            torch.cuda.set_device(self.rank)
+        else:
+            device = torch.device("cpu")
+        model = nn.Sequential(*[nn.Linear(10, 10) for _ in range(2)]).to(device)
+        x = torch.randn(2, 10).to(device)
+        self._test_trace_replicate(model, x)
+
+    @with_comms
+    def test_parallel(self):
+        if self.device_type == "cuda":
+            device = torch.device(self.rank)
+            torch.cuda.set_device(self.rank)
+        else:
+            device = torch.device("cpu")
```

Comment:
nit: this should all be done within the `@with_comms` wrapper, so simply `model = Model().to(self.device_type)` should work.